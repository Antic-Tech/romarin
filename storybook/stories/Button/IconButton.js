import PropTypes from 'prop-types';
import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { MaterialIcons } from 'react-native-vector-icons';

import theme from '../../theme';

const styles = StyleSheet.create({
	button: {
		margin: theme.marginSmall,
		shadowColor: 'rgba(0,0,0,.4)',
		shadowOffset: { height: 1, width: 1 },
		shadowOpacity: 1,
		shadowRadius: 1,
		alignItems: 'center',
		justifyContent: 'center'
	},
	icon: {
		backgroundColor: 'transparent'
	}
});

const IconButton = props => {
	const buttonSizes = {
		xlarge: theme.buttonSizeXLarge,
		large: theme.buttonSizeLarge,
		medium: theme.buttonSizeMedium,
		small: theme.buttonSizeSmall,
		xsmall: theme.buttonSizeXSmall
	};
	const iconSizes = {
		xlarge: theme.iconSizeXLarge,
		large: theme.iconSizeLarge,
		medium: theme.iconSizeMedium,
		small: theme.iconSizeSmall,
		xsmall: theme.iconSizeXSmall
	};
	const { name, size, color, onPress, backgroundColor } = props;

	return (
		<View>
			<TouchableOpacity
				style={[
					styles.button,
					{
						borderRadius: buttonSizes[size],
						height: buttonSizes[size],
						width: buttonSizes[size],
						backgroundColor: theme[backgroundColor]
					}
				]}
				onPress={onPress}
			>
				<MaterialIcons style={styles.icon} size={iconSizes[size]} name={name} color={theme[color]} />
			</TouchableOpacity>
		</View>
	);
};

IconButton.propTypes = {
	name: PropTypes.string,
	size: PropTypes.string,
	color: PropTypes.string,
	backgroundColor: PropTypes.string,
	onPress: PropTypes.func
};

IconButton.defaultProps = {
	name: 'android',
	size: 'medium',
	color: 'colorWhite',
	backgroundColor: 'colorPrimary',
	onPress: () => {}
};

export { IconButton as default };
