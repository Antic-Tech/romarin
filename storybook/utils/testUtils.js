import { mount, shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

export const renderComponent = (jsx, { shallowRender = true, store } = {}) => {
	const renderOptions = { context: { store } };

	if (shallowRender) {
		return shallow(jsx, renderOptions);
	}

	return mount(jsx, renderOptions);
};

export const componentFactory = Component => (props = {}, options = {}) => {
	return renderComponent(<Component {...props} />, options);
};

export const getTree = jsx => renderer.create(jsx).toJSON();

export const verifySnapshot = jsx => {
	const tree = getTree(jsx);

	expect(tree).toMatchSnapshot();
};
