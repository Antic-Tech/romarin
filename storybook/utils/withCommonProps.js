import React from 'react';
import { StyleSheet } from 'react-native';
import ThemeProvider from '../theme';

const Sizes = {
	XXSmall: 'XXSmall',
	XSmall: 'XSmall',
	Small: 'Small',
	Medium: 'Medium',
	Large: 'Large',
	XLarge: 'XLarge',
	XXLarge: 'XXLarge'
};

const withPadding = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				padding,
				paddingHorizontal,
				paddingVertical,
				paddingTop,
				paddingBottom,
				paddingLeft,
				paddingRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					padding: padding ? ThemeProvider.theme[`corePadding${padding}`] : undefined,
					paddingHorizontal: paddingHorizontal
						? ThemeProvider.theme[`corePadding${paddingHorizontal}`]
						: undefined,
					paddingVertical: paddingVertical ? ThemeProvider.theme[`corePadding${paddingVertical}`] : undefined,
					paddingTop: paddingTop ? ThemeProvider.theme[`corePadding${paddingTop}`] : undefined,
					paddingBottom: paddingBottom ? ThemeProvider.theme[`corePadding${paddingBottom}`] : undefined,
					paddingLeft: paddingLeft ? ThemeProvider.theme[`corePadding${paddingLeft}`] : undefined,
					paddingRight: paddingRight ? ThemeProvider.theme[`corePadding${paddingRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withMargin = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				margin,
				marginHorizontal,
				marginVertical,
				marginTop,
				marginBottom,
				marginLeft,
				marginRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					margin: margin ? ThemeProvider.theme[`coreMargin${margin}`] : undefined,
					marginHorizontal: marginHorizontal
						? ThemeProvider.theme[`coreMargin${marginHorizontal}`]
						: undefined,
					marginVertical: marginVertical ? ThemeProvider.theme[`coreMargin${marginVertical}`] : undefined,
					marginTop: marginTop ? ThemeProvider.theme[`coreMargin${marginTop}`] : undefined,
					marginBottom: marginBottom ? ThemeProvider.theme[`coreMargin${marginBottom}`] : undefined,
					marginLeft: marginLeft ? ThemeProvider.theme[`coreMargin${marginLeft}`] : undefined,
					marginRight: marginRight ? ThemeProvider.theme[`coreMargin${marginRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withCommonProps = (Component) => withPadding(withMargin(Component));

export { withPadding, withMargin, withCommonProps };
