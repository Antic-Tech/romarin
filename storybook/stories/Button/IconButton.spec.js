import React from 'react';
import IconButton from './IconButton';

import { verifySnapshot } from '../../utils/testUtils';

jest.mock('react-native-vector-icons', () => ({ MaterialIcons: () => null }));

describe('IconButton', () => {
	describe('Snapshots', () => {
		it('renders correctly', () => {
			verifySnapshot(<IconButton />);
		});
	});
});
