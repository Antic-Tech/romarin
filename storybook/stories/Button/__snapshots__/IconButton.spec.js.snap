// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Snapshots renders correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2C7CF6",
        "borderRadius": 68,
        "height": 68,
        "justifyContent": "center",
        "margin": 8,
        "opacity": 1,
        "shadowColor": "rgba(0,0,0,.4)",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
        "width": 68,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  />
</View>
`;
